lib/cmds/template_generator.js

```diff
@@ -6,22 +6,38 @@ class TemplateGenerator {
     this.templateName = templateName;
   }
 
+  downloadAndGenerate(uri, destinationFolder, name) {
+    const self = this;
+    let {url, filePath} = this.getExternalProject(uri);
+    let tmpFilePath = fs.tmpDir(filePath);
+    console.log(__('Installing Template from ' + uri + '....').green);
+
+    fs.mkdirpSync(utils.dirname(tmpFilePath));
```

Comment:
Since every other call is async, you should maybe use the async dir creation too, maybe with a waterfall to organize